lib/puppet/pops/lookup/lookup_adapter.rb

```diff
@@ -417,11 +417,11 @@ def initialize_env_provider(lookup_invocation)
       elsif ep_config.version >= 5
         unless provider_name.nil? || Puppet[:strict] == :off
           Puppet.warn_once(:deprecation, 'environment.conf#data_provider',
-            "Defining environment_data_provider='#{provider_name}' in environment.conf is deprecated", env_path + 'environment.conf')
+            _("Defining environment_data_provider='%{provider_name}' in environment.conf is deprecated") % { provider_name: provider_name }, env_path + _('environment.conf'))
 
           unless provider_name == 'hiera'
             Puppet.warn_once(:deprecation, 'environment.conf#data_provider_overridden',
-              "The environment_data_provider='#{provider_name}' setting is ignored since '#{config_path}' version >= 5", env_path + 'environment.conf')
+              _("The environment_data_provider='%{provider_name}' setting is ignored since '%{config_path}' version >= 5") % { provider_name: provider_name, config_path: config_path }, env_path + _('environment.conf'))
```

Comment:
I don't think we want to externalize `'environment.conf'` here and in the preceding lines because they combine to form a file system path.